src/dist/failure_detector/failure_detector.cpp

```diff
@@ -69,11 +69,16 @@ error_code failure_detector::start(
     _grace_milliseconds = grace_seconds * 1000;
 
     _use_allow_list   = use_allow_list;
+    _primary_address  = primary_address();
```

Comment:
use _local_address instead, as the sub class in meta server etc. has "primary_address" whch has differ